### PR TITLE
ci(sync): increase the height of blocks for some full sync jobs

### PR DESCRIPTION
**Pull request by arya2**
## Motivation

This is the standard fix for when a full sync job runs too long (`logs-checkpoint` needs to sync ~45k blocks as of the latest checkpoint update).

## Solution

- Replace full sync job for 1790k blocks with one for 1800k blocks
- Add full sync job for 1820k blocks

See [fixing-ci-sync-timeouts](https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/continuous-integration.md#fixing-ci-sync-timeouts)

## Review

Anyone can review.

### Reviewer Checklist

  - [x] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [x] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

**Comment by arya2** [comment]
The checkpoint step took 4-5 hours to get to [1810k blocks](https://github.com/ZcashFoundation/zebra/actions/runs/3223410431/jobs/5291373001#step:6:812) (timed out after ~1817k). Adding just one extra step this time.

**Comment by arya2** [comment]
ssh failure in `logs-checkpoint`:

> Timeout, server 34.123.232.71 not responding.
> 
> ..
> 
> ERROR: (gcloud.compute.ssh) [/usr/bin/ssh] exited with return code [255].

https://github.com/ZcashFoundation/zebra/actions/runs/3236318547/jobs/5302131447#step:6:777

**Comment by teor2345** [comment]
> ssh failure in `logs-checkpoint`: https://github.com/ZcashFoundation/zebra/actions/runs/3236318547/jobs/5302131447#step:6:777

There are lots of different ways `ssh` can fail, can you please quote the actual failure log?

> Timeout, server 34.123.232.71 not responding.

**Review by teor2345 — CHANGES_REQUESTED** [review]
I also saw these sync warnings in the logs:
> 2022-10-11T03:59:58.369043Z  WARN {net="Main"}: zebrad::components::sync::progress: chain updates have stalled, state height has not increased for 30 minutes. Hint: check your network connection, and your computer clock and time zone sync_percent=31.016% current_height=Height(572486) network_upgrade=Sapling time_since_last_state_block=30m target_block_spacing=PT75S max_block_spacing=None is_syncer_stopped=false

https://github.com/ZcashFoundation/zebra/actions/runs/3223410431/jobs/5295854595#step:6:2844

Every time they happen, the sync takes half an hour longer waiting for a single block. So maybe we could drop the timeout back down to 10 minutes first?

(We might still need this PR.)

**Comment by teor2345** [comment]
> Every time they happen, the sync takes half an hour longer waiting for a single block. So maybe we could drop the timeout back down to 10 minutes first?

Let's see how #5397 goes, then try this one?

There is also a timeout in the Rust test we might need to fix.

**Comment by arya2** [comment]
> I also saw these sync warnings in the logs:
> 
> > 2022-10-11T03:59:58.369043Z  WARN {net="Main"}: zebrad::components::sync::progress: chain updates have stalled, state height has not increased for 30 minutes. Hint: check your network connection, and your computer clock and time zone sync_percent=31.016% current_height=Height(572486) network_upgrade=Sapling time_since_last_state_block=30m target_block_spacing=PT75S max_block_spacing=None is_syncer_stopped=false
> 
> https://github.com/ZcashFoundation/zebra/actions/runs/3223410431/jobs/5295854595#step:6:2844
> 
> Every time they happen, the sync takes half an hour longer waiting for a single block. So maybe we could drop the timeout back down to 10 minutes first?

We should definitely fix that first.

> (We might still need this PR.)

It's doubtful it'll make it from 1790k to ~1834k, though we might be able to replace `logs-1790k` (~2h 40m) with `logs-1800k` (seems to be another ~2h 20m) instead of adding a new step.

**Comment by gustavovalverde** [comment]
@arya2 do we still need this PR?

**Comment by gustavovalverde** [comment]
Seems like we are still having timeouts in the `main` branch, but I'll left @teor2345 the last call here even though is a standard change, as I'm not sure if there are other works in progress to fix it another way

**Comment by arya2** [comment]
> do we still need this PR?

I believe we do, but I'll just quickly check the logs for the timeouts and perhaps replace `logs-1790k` with `logs-1800k`.

**Comment by teor2345** [comment]
> Seems like we are still having timeouts in the `main` branch, but I'll left @teor2345 the last call here even though is a standard change, as I'm not sure if there are other works in progress to fix it another way

We're done with the fixes due to PR #5257, it should all be standard fixes from now on.

**Review by gustavovalverde — DISMISSED** [review]
LGTM

File: .github/workflows/deploy-gcp-tests.yml [hunk 1]
```diff
@@ -962,16 +962,80 @@ jobs:
           ${{ inputs.test_id }} | \
           tee --output-error=exit /dev/stderr | \
           grep --max-count=1 --extended-regexp --color=always \
-          -e 'estimated progress.*current_height.*=.*179[0-9][0-9][0-9][0-9].*remaining_sync_blocks' \
+          -e 'estimated progress.*current_height.*=.*180[0-9][0-9][0-9][0-9].*remaining_sync_blocks' \
```
Comment (teor2345) [inline .github/workflows/deploy-gcp-tests.yml]: This line is actually redundant, 1,800,000 - 1,809,999 is covered by the `18[0-9][0-9][0-9][0-9][0-9]` regex.

```suggestion
```

File: .github/workflows/deploy-gcp-tests.yml [hunk 2]
```diff
@@ -962,16 +962,80 @@ jobs:
...
+          --command \
+          "\
+          sudo docker logs \
+          --tail all \
+          --follow \
+          ${{ inputs.test_id }} | \
+          tee --output-error=exit /dev/stderr | \
+          grep --max-count=1 --extended-regexp --color=always \
+          -e 'estimated progress.*current_height.*=.*182[0-9][0-9][0-9][0-9].*remaining_sync_blocks' \
+          -e 'estimated progress.*current_height.*=.*1[8-9][2-9][0-9][0-9][0-9][0-9].*remaining_sync_blocks' \
```
Comment (teor2345) [inline .github/workflows/deploy-gcp-tests.yml]: These regexes look like they might be incorrect, they seem to be skipping 1,900,000 - 1,919,999.

We want to match:
- 1,820,000 - 1,899,999
- 1,900,000 - 1,999,999

```suggestion
          -e 'estimated progress.*current_height.*=.*18[2-9][0-9][0-9][0-9][0-9].*remaining_sync_blocks' \
          -e 'estimated progress.*current_height.*=.*19[0-9][0-9][0-9][0-9][0-9].*remaining_sync_blocks' \
```

**Review by teor2345 — CHANGES_REQUESTED** [review]
I think the regexes might be wrong here?

**Comment by mpguerra** [comment]
is there a github issue that this is related to that I can link in?

**Comment by arya2** [comment]
> is there a github issue that this is related to that I can link in?

No, this was ad-hoc and trivial.

**Review by teor2345 — APPROVED** [review]
Thanks!

**Comment by teor2345** [comment]
@mergifyio update

**Pull request merged**